### PR TITLE
chore: Migrate user `role` to new column

### PR DESCRIPTION
**Pull request by tommoor**
In order to support a new `guest` role first we need to transition the existing booleans to an enum.

### This PR
- [x] Add new database column
- [x] Double write to existing and new field
- [x] Backfill rows

### Future PR's
- [x] Change `isViewer` and `isAdmin` to getters
- [ ] Remove old columns
- [ ] Make `role` not nullable

towards #2946

### CODE REVIEW

File: server/migrations/20240327015248-add-user-role.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+module.exports = {
+  async up (queryInterface, Sequelize) {
+    await queryInterface.addColumn("users", "role", {
+      type: Sequelize.ENUM("admin", "member", "viewer", "guest"),
```
Comment (tommoor) [inline server/migrations/20240327015248-add-user-role.js:6]: ENUM's are annoying to migrate so including `guest` here from the beginning, although it won't be used yet.

File: server/migrations/20240327015248-add-user-role.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      allowNull: true,
+    });
+    if (process.env.DEPLOYMENT === "hosted") {
+      return;
+    }
+
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'viewer' WHERE "isViewer" = true`,
+      {
+        type: queryInterface.sequelize.QueryTypes.SELECT,
```
Comment (apoorv-mishra) [inline server/migrations/20240327015248-add-user-role.js:16]:
```suggestion
        type: queryInterface.sequelize.QueryTypes.UPDATE,
```

File: server/migrations/20240327015248-add-user-role.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      }
+    );
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'admin' WHERE "isAdmin" = true`,
+      {
+        type: queryInterface.sequelize.QueryTypes.SELECT,
```
Comment (apoorv-mishra) [inline server/migrations/20240327015248-add-user-role.js:22]:
```suggestion
        type: queryInterface.sequelize.QueryTypes.UPDATE,
```

File: server/migrations/20240327015248-add-user-role.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      }
+    );
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'member' WHERE role IS NULL`,
+      {
+        type: queryInterface.sequelize.QueryTypes.SELECT,
```
Comment (apoorv-mishra) [inline server/migrations/20240327015248-add-user-role.js:28]:
```suggestion
        type: queryInterface.sequelize.QueryTypes.UPDATE,
```

File: server/migrations/20240327015248-add-user-role.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      }
+    );
```
Comment (apoorv-mishra) [inline server/migrations/20240327015248-add-user-role.js:30]: Better to wrap these updates within a transaction?

File: server/migrations/20240327015248-add-user-role.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'admin' WHERE "isAdmin" = true`,
+      {
+        type: queryInterface.sequelize.QueryTypes.SELECT,
+      }
+    );
+    await queryInterface.sequelize.query(
+      `UPDATE users SET role = 'member' WHERE role IS NULL`,
```
Comment (apoorv-mishra) [inline server/migrations/20240327015248-add-user-role.js:26]: Can impose a non-null constraint on `role` through a follow up query.

Comment (tommoor) [reply server/migrations/20240327015248-add-user-role.js:26]: Will add in later PR to allow for async backfill in cloud

**Pull request merged**